src/app/App.tsx

```diff
@@ -106,19 +115,44 @@ function App(): JSX.Element {
     // finds the correct topic and adds a need on top of its needList
     setTopics(prev =>
       prev.map(topic =>
-        topic.content.id === topicId
-          ? {
+        topic.content.id !== topicId
+          ? topic
+          : {
               ...topic,
               content: {
                 ...topic.content,
                 needs: [newNeed, ...topic.content.needs],
               },
             }
-          : topic
       )
     )
   }
 
+  const handleNeedUpvote =
+    (topicId: string) => (needId: string) => (newUpvotes: number) => {
+      // finds the relevant Topic, inside it finds the relevant need and updates it upvote count
+      setTopics(prev =>
+        prev.map(topic =>
+          topic.content.id !== topicId
+            ? topic
+            : {
+                ...topic,
+                content: {
+                  ...topic.content,
+                  needs: topic.content.needs.map(need =>
+                    need.id !== needId
+                      ? need
+                      : {
+                          ...need,
+                          upvotes: newUpvotes,
+                        }
+                  ),
+                },
+              }
+        )
+      )
```

Comment:
This could be written a bit more readable by having a few more variables like `correspondingTopic` or `newNeed` etc.

Reply:
Good Point I will try it out!